analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.Arithmetic.cs

```diff
@@ -249,6 +249,20 @@ private static NumberConstraint AccountForZero(NumberConstraint constraint)
                 return 0;
             }
         }
+        else if ((left.IsPositive || right.IsNegative) && left.Max.HasValue && right.Min.HasValue)
```

Comment:
The code smell on this method suggests that the complexity of it should be reduced to <= 15, and I agree that the function is now difficult to follow due to the many branches and the logic operators, sometimes nested.

I think giving semantics to the following conditions, via local functions, would help:
- `(left.IsNegative || right.IsPositive) && left.Min.HasValue && right.Max.HasValue`
- `left.Min.HasValue && left.Max.HasValue && right.Min.HasValue && right.Max.HasValue`

Extracting nested if blocks on the first two branches would also help (in the first version of this code I had `CanOverlap` and a more concise version of this, which didn't have nested conditional statements on the same variables).

Local functions are not included in the calculation of the cognitive complexity of the parent method.